### PR TITLE
Update Event-Buffer functionality with changes to StreamBuffer & Stre…

### PR DESCRIPTION
**Pull request by BenNzewi**
…amStatus tables to support EventIndexer

### CODE REVIEW

File: event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/manager/CatchupEventBufferProcessor.java [hunk 1]
```diff
@@ -27,12 +27,12 @@
     private InterceptorChainProcessorProducer interceptorChainProcessorProducer;
```
Comment (allanmckenzie) [inline event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/manager/CatchupEventBufferProcessor.java]: I think this class (CatchupEventBufferProcessor) is missing a test

Comment (BenNzewi) [reply event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/manager/CatchupEventBufferProcessor.java]: CatchupEventBufferProcessorTest added

File: pom.xml [hunk 2]
```diff
@@ -42,8 +42,8 @@
         <utilities.version>1.17.0-M2</utilities.version>
         <test-utils.version>1.22.0-M1</test-utils.version>
         <framework.version>6.0.0-M20</framework.version>
-        <common-bom.version>1.28.0</common-bom.version>
-        <framework-api.version>4.0.0-M13</framework-api.version>
+        <common-bom.version>1.29.0</common-bom.version>
```
Comment (mapingo) [inline pom.xml]: Can the common-bom version be updated to 2.0.2

Comment (BenNzewi) [reply pom.xml]: common-bom updated to 2.0.2 with latest push.

File: event-buffer/event-buffer-liquibase/src/main/resources/liquibase/event-buffer-changesets/004-add-component-to-stream-buffer.changelog.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                newColumnName="position"
+                tableName="stream_buffer"
+        />
+
+        <addColumn
+                tableName="stream_buffer">
+            <column
+                    name="component"
+                    type="varchar(100)"
+                    defaultValue="event_listener">
```
Comment (arcadius) [inline event-buffer/event-buffer-liquibase/src/main/resources/liquibase/event-buffer-changesets/004-add-component-to-stream-buffer.changelog.xml]: Should this be upper case or ?

Comment (BenNzewi) [reply event-buffer/event-buffer-liquibase/src/main/resources/liquibase/event-buffer-changesets/004-add-component-to-stream-buffer.changelog.xml]: updated to EVENT_LISTENER in latest push

**Pull request merged**